### PR TITLE
ORC-815: Build and test C++ module on CLang12

### PR DESCRIPTION
**Pull request by williamhyun**
### What changes were proposed in this pull request?
This PR aims to build and test the C++ module on CLang12

### Why are the changes needed?
Currently, the build fails with new warnings. 

### How was this patch tested?
Manually built and tested.
```
$ make test-out
Test project /Users/william/orc/build
    Start 1: orc-test
1/2 Test #1: orc-test .........................   Passed    3.99 sec
    Start 2: tool-test
2/2 Test #2: tool-test ........................   Passed   11.96 sec

100% tests passed, 0 tests failed out of 2
```

### CODE REVIEW

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. (Pending CI).

**Comment by dongjoon-hyun** [comment]
Oh, it seems that old compilers complains about this.
```
error: unknown warning option '-Wno-suggest-override'; did you mean '-Wno-shift-overflow'? [-Werror,-Wunknown-warning-option]
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -98,7 +98,7 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
   set (WARN_FLAGS "${WARN_FLAGS} -Wno-gnu-zero-variadic-macro-arguments")
   set (WARN_FLAGS "${WARN_FLAGS} -Wconversion")
   if (CMAKE_HOST_APPLE AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "11.0")
-    set (WARN_FLAGS "${WARN_FLAGS} -Wno-c++2a-compat")
+    set (WARN_FLAGS "${WARN_FLAGS} -Wno-c++2a-compat -Wno-suggest-override -Wno-suggest-destructor-override")
   endif ()
```
Comment (dongjoon-hyun) [inline CMakeLists.txt:102]: Maybe, shall we change like this?
```
-  if (CMAKE_HOST_APPLE AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "11.0")
+  if (CMAKE_HOST_APPLE AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "12.0")
+    set (WARN_FLAGS "${WARN_FLAGS} -Wno-c++2a-compat -Wno-suggest-override -Wno-suggest-destructor-override")
+  elseif (CMAKE_HOST_APPLE AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "11.0")
```

**Comment by dongjoon-hyun** [comment]
I updated your branch to see the result. If it fails still, you can revert it, @williamhyun ~

**Comment by dongjoon-hyun** [comment]
Please review and merge this if CI passed, @williamhyun . 😄

**Comment by williamhyun** [comment]
+1 LGTM, thank you for the edits.

**Comment by dongjoon-hyun** [comment]
Thank you, @williamhyun . I tested and backported this to branch-1.6 on M1.
```
cmake .. -DBUILD_JAVA=OFF
make package
make test-out
Test project /Users/dongjoon/APACHE/orc-merge/build
    Start 1: orc-test
1/2 Test #1: orc-test .........................   Passed    3.22 sec
    Start 2: tool-test
2/2 Test #2: tool-test ........................   Passed    9.87 sec

100% tests passed, 0 tests failed out of 2

Total Test time (real) =  13.10 sec
Built target test-out
```

**Comment by dongjoon-hyun** [comment]
I'll cherry-pick to branch-1.5 for Apache ORC 1.5.13.

**Pull request closed without merging**